README.md

```diff
@@ -211,6 +213,23 @@ $records = $porter->import(
         ->setCacheAdvice(CacheAdvice:SHOULD_CACHE())
 );
 ```
+### Cache key
+
+The cache key can optionally be generated by an implementation of `CacheKeyGeneratorInterface` passed into `CachingConnector::fetch`. This implementation should provide one method `generateCacheKey` which returns a [PSR-6][PSR-6]-compatible cache key.
+
+The default implementation `SourceAndOptionsHashCacheKeyGenerator` generates keys comprised of the source and options parameters passed to `Connector::fetch`. Options are sorted before the cache key is created so the order of options are insignificant.
+
+#### Implementation example
+
+```php
+class MyCacheKeyGenerator implements CacheKeyGenerator
+{
+    public function generateCacheKey($source, array $optionsSorted)
+    {
+        return str_replace(str_split('{}()/\@:'), '.', json_encode([$source, $optionsSorted], JSON_UNESCAPED_SLASHES));
```

Comment:
This example is too complicated because details of PSR-6 compliance and specific `json_encode` options are just muddying the point of the example, which should be just to compose a cache key with the fewest, simplest steps possible.

Consider the following implementation which does not introduce any unnecessary complexity or noise into the example.

```php
return array_reduce([$source] + $sortedOptions, function ($key, $option) { return "$key$option"; });
```

We can also add a lead description for the example, e.g. *The following example demonstrates a simple cache key generation implementation using string concatenation.*

Of course, one weakness of this example is its assumption that all options can be cast to string. If we were going to keep the JSON example it should probably look something like the following.

```php
return json_encode([$source, $sortedOptions]);
```

It is acceptable to use naive and impractical examples in order to achieve simplicity and better convey the fundamental idea to the reader, free of distracting details.